dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/QuarterlyPeriodType.java

```diff
@@ -180,6 +180,11 @@ public Period getPreviousPeriod( Period period, org.hisp.dhis.calendar.Calendar
     @Override
     public String getIsoDate( DateTimeUnit dateTimeUnit, org.hisp.dhis.calendar.Calendar calendar  )
     {
+
+        if (calendar.name() == "persian" && dateTimeUnit.isIso8601()) {

```

Comment:
This looks a bit hacky. Can we make it more dynamic?

Also please follow Java code style.